Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -103,28 +97,39 @@ void CookieStore::get(CookieStoreGetOptions&& options, Ref<DeferredPromise>&& pr
         return;
     }
 
-    if (!m_cookieJar) {
-        promise->reject(SecurityError);
-        return;
-    }
-
-    auto& document = *downcast<Document>(context);
-    auto completionHandler = [promise = WTFMove(promise)] (std::optional<Vector<Cookie>>&& cookies) {
-        if (!cookies) {
-            promise->reject(TypeError);
+    auto getCookies = [options = WTFMove(options), promise = WTFMove(promise)](ScriptExecutionContext& context) {
+        auto& document = downcast<Document>(context);
+        auto* page = document.page();
+        if (!page) {
+            promise->reject(SecurityError);
             return;
         }
 
-        auto& cookiesVector = *cookies;
-        if (cookiesVector.isEmpty()) {
-            promise->resolveWithJSValue(JSC::jsNull());
-            return;
-        }
+        auto& cookieJar = page->cookieJar();
+        auto completionHandler = [promise = WTFMove(promise)] (std::optional<Vector<Cookie>>&& cookies) {
+            if (!cookies) {
+                promise->reject(TypeError);
+                return;
+            }
+
+            auto& cookiesVector = *cookies;
+            if (cookiesVector.isEmpty()) {
+                promise->resolveWithJSValue(JSC::jsNull());
+                return;
+            }
 
-        promise->resolve<IDLDictionary<CookieListItem>>(CookieListItem(WTFMove(cookiesVector[0])));
+            promise->resolve<IDLDictionary<CookieListItem>>(CookieListItem(WTFMove(cookiesVector[0])));
```

Comment:
We would need to hop back to the worker thread here before resolving the promise.

Also why only the first item? Should we specialise the cookie jar to only return 1 item?